lib/picos_sync/picos_sync.mli

```diff
@@ -158,6 +158,80 @@ module Lazy : sig
       ]} *)
 end
 
+module Event : sig
+  (** An implementation of first-class synchronous communication for {!Picos}.
+
+      Events describe a thing that might happen in the future, or a concurrent
+      offer or request that might be accepted or succeed, but is cancelable if
+      some other event happens first.
+
+      ℹ️ This module intentionally mimics the
+      {{:https://ocaml.org/manual/5.2/api/Event.html} [Event]} module provided
+      by the OCaml POSIX threads library. *)
+
+  type !'a t
+  (** An event returning a value of type ['a]. *)
+
+  type 'a event = 'a t
+  (** An alias for the {!Event.t} type to match the
+      {{:https://ocaml.org/manual/5.2/api/Event.html} [Event]} module
+      signature. *)
+
+  (** {2 Composing events} *)
+
+  val choose : 'a t list -> 'a t
+  (** [choose events] return an event that offers all of the given events and
+      then commits to at most one of them. *)
+
+  val wrap : 'b t -> ('b -> 'a) -> 'a t
+  (** [wrap event fn] returns an event that acts as the given [event] and then
+      applies the given function to the value in case the event is committed
+      to. *)
+
+  val map : ('b -> 'a) -> 'b t -> 'a t
+  (** [map fn event] is equivalent to {{!wrap} [wrap event fn]}. *)
+
+  val guard : (unit -> 'a t) -> 'a t
+  (** [guard thunk] returns an event that, when {{!sync} synchronized}, calls
+      the [thunk], and then behaves like the resulting event. *)
+
+  (** {2 Consuming events} *)
+
+  val sync : 'a t -> 'a
```

Comment:
In addition to the `timeout` event constructor, we could also provide an optional timeout here.  Perhaps using an optional `timeoutf` argument like in Kcas.  The timeout would then cause a `Timeout` exception to be raised, which would need to be handled separately.